internal/semver/semver.go

```diff
@@ -0,0 +1,64 @@
+package semver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+)
+
+func getTargetVaultVersion(ctx context.Context, client *api.Client) (string, error) {
+	resp, err := client.Sys().SealStatusWithContext(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	if resp == nil {
+		return "", fmt.Errorf("expected response data, got nil response")
+	}
+
+	if resp.Version == "" {
+		return "", fmt.Errorf("key %q not found in response", consts.FieldVersion)
+	}
+
+	return resp.Version, nil
+}
+
+// GreaterThanOrEqual receives a context, a Vault API client
+// and a minimum version that the Vault server version
+// should be above.
+//
+// It uses the go-version package
+// to perform a semantic version comparison, and
+// returns:
+//    - a boolean describing whether the Vault
+//      server version was above the minimum version
+//    - the current Vault server version as a string
+//    - errors captured during operation, if any
+//
+// This function can be used to perform semantic version comparisons
+// to conditionally enable features, or to resolve any diffs in the TF
+// state based on the Vault version.
+func GreaterThanOrEqual(ctx context.Context, client *api.Client, minVersionString string) (bool, string, error) {
```

Comment:
It looks like we will be calling the API for every Version check, this seems like a unnecessary increase in the overall the number of requests per operation. It is very unlikely that the vault server version will change during a run of Terraform.